observability-lib/cmd/builder.go

```diff
@@ -35,60 +34,75 @@ const (
 
 type BuildOptions struct {
 	Name              string
-	FolderUID         string
 	Platform          grafana.TypePlatform
 	TypeDashboard     TypeDashboard
 	MetricsDataSource *grafana.DataSource
 	LogsDataSource    *grafana.DataSource
-	AlertsTags        map[string]string
-	SlackWebhookURL   string
-	SlackToken        string
 	SlackChannel      string
+	SlackWebhookURL   string
+	AlertsTags        map[string]string
 }
 
-func Build(options *BuildOptions) (*grafana.Dashboard, error) {
-	var db *grafana.Dashboard
-	var err error
-
-	dashboardOptions := &grafana.DashboardOptions{
-		Name:              options.Name,
-		MetricsDataSource: options.MetricsDataSource,
-		FolderUID:         options.FolderUID,
-		Platform:          options.Platform,
-		AlertsTags:        options.AlertsTags,
-		SlackWebhookURL:   options.SlackWebhookURL,
-		SlackToken:        options.SlackToken,
-		SlackChannel:      options.SlackChannel,
-	}
-
+func BuildDashboardWithType(options *BuildOptions) (*grafana.Dashboard, error) {
 	switch options.TypeDashboard {
 	case TypeDashboardCoreNode:
-		db, err = corenode.NewDashboard(dashboardOptions)
+		return corenode.NewDashboard(&corenode.Props{
```

Comment:
exploring: the `Props` (corenode, corenodecomponents, k8sresources) seem to share a fair amount of fields. Could the Props type be consolidated with validation added in NewDashboard to ensure the right fields are passed in? 

I do like that the right fields are asserted at build time though, so its a tradeoff.

Reply:
The exported function NewDashboard from each modules needs to be easily usable from anywhere without using cmd. I'm open for idea on the params needed for the NewDashboard function exported from each module. For now I don't enforce through the library the common params when building dashboards and alerts.